src/provider/provider.go

```diff
@@ -0,0 +1,97 @@
+package provider
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	logger "github.com/sirupsen/logrus"
+
+	"github.com/envoyproxy/ratelimit/src/config"
+)
+
+type Provider struct {
+	Endpoint     string
+	Subpath      []string
+	PollInterval time.Duration
+	PollTimeout  time.Duration
+	ConfigChan   chan []config.RateLimitConfigToLoad
+	httpClient   *http.Client
+	StopSig      bool
```

Comment:
StopFlag?

Reply:
Switch to the channel to implement the stop behavior.